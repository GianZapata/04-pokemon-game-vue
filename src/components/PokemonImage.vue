<template>
  <div class="my-20">
    <img
      class="w-64 h-64"
      :src="imgSrc"
      alt="Pokemon"
      :class="showPokemon ? 'show' : 'brightness-0 disable-drag'"
    />
  </div>
</template>
<script lang="ts">
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    },
  },
}
</script>
<style lang="css">
.disable-drag {
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
</style>
