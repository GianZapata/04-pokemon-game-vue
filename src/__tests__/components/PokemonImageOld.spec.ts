import { describe, test, expect } from "vitest"
import { shallowMount } from "@vue/test-utils"
import PokemonImage from "@/components/PokemonImageOld.vue"

describe("PokemonImage Component", () => {
  test("Debe de hacer el snapshot", () => {
    const wrapper = shallowMount(PokemonImage, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test("Debe de mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = shallowMount(PokemonImage, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    })

    const img = wrapper.find("img")

    expect(img.exists()).toBe(true)
    expect(img.classes("hide-pokemon")).toBe(true)
  })

  test("Debe de mostrar la imagen del showPokemon:true", () => {
    const pokemonURL =
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"

    const wrapper = shallowMount(PokemonImage, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    })

    const img = wrapper.find("img")
    const src = img.attributes("src")

    expect(img.exists()).toBe(true)
    expect(img.classes("hide-pokemon")).toBe(false)
    expect(img.classes("show-pokemon")).toBe(true)
    expect(src).toBe(pokemonURL)
  })
})
