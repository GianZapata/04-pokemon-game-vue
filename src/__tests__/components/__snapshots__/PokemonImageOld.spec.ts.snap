// Vitest Snapshot v1

exports[`PokemonImage Component > Debe de hacer el snapshot 1`] = `"<div class=\\"my-20\\"><img class=\\"w-64 h-64 fade-in hide-pokemon brightness-0 disable-drag\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" alt=\\"Pokemon\\"></div>"`;
